core/src/main/java/org/elasticsearch/ingest/core/Pipeline.java

```diff
@@ -83,9 +85,10 @@ public String getDescription() {
     public final static class Factory {
 
         public Pipeline create(String id, Map<String, Object> config, Map<String, Processor.Factory> processorRegistry) throws Exception {
+            AtomicInteger processorIdGenerator = new AtomicInteger(0);
```

Comment:
I think there isn't a simple helper class that wraps an integer. So it is ok to use AtomicInteger?